### PR TITLE
docs: Docusaurus v3.0 blog post

### PR DESCRIPTION
**Pull request by slorber**
## Motivation

Docusaurus v3 blog post

https://deploy-preview-9433--docusaurus-2.netlify.app/blog/releases/3.0

### Links

- Release PR: https://github.com/facebook/docusaurus/pull/9478
- Blog prod: https://docusaurus.io/blog/releases/3.0
- Blog PR: https://github.com/facebook/docusaurus/pull/9433
- Blog preview: https://deploy-preview-9433--docusaurus-2.netlify.app/blog/releases/3.0
- Twitter thread draft: https://app.feedhive.com/public/post/MlD1NgEvJC

### CODE REVIEW

File: website/blog/releases/3.0/index.mdx [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+
+We initially planned to release more frequent major versions, but this one has taken longer than expected. **Upgrading to [MDX v2](https://mdxjs.com/blog/v2/) is the main challenge** to the adoption of this new version. We went the extra mile to make this upgrade as easy as possible, notably by adding [compatibility options for MDX v1](/docs/3.0.0-rc.0/api/docusaurus-config#markdown).
+
+The simplest sites will only need to upgrade a few npm dependencies. For more complex sites, we came up with a few strategies than can help you upgrade more confidently:
+
+- [prepare your site](/blog/preparing-your-site-for-docusaurus-v3) ahead of time, incrementally, while staying under Docusaurus v2
+- [set up visual regression tests](/blog/upgrading-frontend-dependencies-with-confidence-using-visual-regression-testing) to catch unexpected visual changes happening during the upgrade
```
Comment (Josh-Cena) [inline website/blog/releases/3.0/index.mdx:24]:
```suggestion
- [Prepare your site](/blog/preparing-your-site-for-docusaurus-v3) ahead of time, incrementally, while staying on Docusaurus v2
- [Set up visual regression tests](/blog/upgrading-frontend-dependencies-with-confidence-using-visual-regression-testing) to catch unexpected visual changes happening during the upgrade
```

File: website/blog/releases/3.0/index.mdx [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+
+This section only gives you a quick glance. All the breaking changes are thoroughly documented on the [**v3 upgrade guide**](/docs/3.0.0-rc.0/migration/v3).
+
+Docusaurus v3 upgraded a few dependencies to new major versions, each coming with their own breaking changes:
+
+- Node.js v16 ➡️ v18
+- React v17 ➡️ v18
+- MDX v1 ➡️ v2
+- TypeScript v4 ➡️ v5
+- Prism-React-Renderer v1 ➡️ v2
```
Comment (Josh-Cena) [inline website/blog/releases/3.0/index.mdx:42]:
```suggestion
- prism-react-renderer v1 ➡️ v2
```

**Review by Josh-Cena — COMMENTED** [review]
Looks mostly great! I am already getting excited.

**Comment by Josh-Cena** [comment]
Have we mentioned the maintenance state of v2 onwards anywhere? I think for every new major release, we should restate the note on https://docusaurus.io/community/release-process#versions, because it helps with discoverability. We can add this to the intro section:

> This also means v2 has now entered **maintenance mode**. It will continue to receive support only for major security issues for 3 months (ending in late January, 2024). Otherwise, all features will be frozen and non-critical bugs will not be fixed.
>
> It is recommended to upgrade within that time frame to v3.

File: website/blog/releases/3.0/index.mdx [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+
+If you created custom [MDX plugins](/docs/markdown-features/plugins) (Remark/Rehype), the AST is slightly different, and you might need to refactor them.
+
+:::
+
+This notably enables us to add a [CommonMark mode](/docs/3.0.0-rc.0/markdown-features#mdx-vs-commonmark) that should make it easier for existing documentations to adopt Docusaurus. It is currently opt-in and **experimental** and limited ([some Docusaurus features will not work](https://github.com/facebook/docusaurus/issues/9092)). In Docusaurus v3, all files are still interpreted as MDX, but we plan to interpret `.md` files as CommonMark in an upcoming major version, and recommend to use the `.mdx` extension for any file using JSX or ES modules.
+
+We also introduced a new way to [configure Markdown globally for your site](/docs/3.0.0-rc.0/api/docusaurus-config#markdown), and plan to add more flexible options later.
```
[the remark this answers is not in the document]
Comment (slorber) [reply website/blog/releases/3.0/index.mdx:111]: Once v3 is out I'll remove the version.
Similarly we had a previous blog post referencing beta.0 that I had to temporarily edit to rc.0.

That's a bit annoying, but only temporary

File: website/blog/releases/3.0/index.mdx [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+title: Announcing Docusaurus 3.0
+authors:
+  - slorber
+tags: [release]
+image: ./img/social-card.png
+date: 2023-10-25
+unlisted: true
+---
+
+Today we are happy to **announce Docusaurus 3.0**! 🥳
```
Comment (dmitryvinn) [inline website/blog/releases/3.0/index.mdx:11]: Today, we are happy to **announce Docusaurus 3.0**! 🥳

**Review by Josh-Cena — COMMENTED** [review]
Incomplete reviews

File: website/blog/releases/3.0/index.mdx [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+
+At [**Meta Open Source**](https://opensource.fb.com/), we believe Docusaurus will help you build the **best documentation websites** with **minimal effort**, letting you **focus on what really matters**: writing the content.
+
+This is a new **major version** of Docusaurus, coming with **new exciting features** and upgraded dependencies.
+
+We respect [**Semantic Versioning**](https://semver.org/) and this release also includes **breaking changes** that we documented thoroughly in the [**v3 upgrade guide**](/docs/3.0.0-rc.0/migration/v3). Breaking changes can be annoying, but are necessary to set the ground for a **new wave of Docusaurus features** that we plan to implement later.
```
Comment (dmitryvinn) [inline website/blog/releases/3.0/index.mdx:17]: In line with the [Semantic Versioning](https://semver.org/) principles, this release includes breaking changes we documented thoroughly in the [v3 upgrade guide](https://deploy-preview-9433--docusaurus-2.netlify.app/docs/3.0.0-rc.0/migration/v3/). Breaking changes can be bothersome, but they are necessary to set the ground for a new wave of Docusaurus features we plan to implement.

File: website/blog/releases/3.0/index.mdx [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+
+![v3.0 social-card image](./img/social-card.png)
+
+{/* truncate */}
+
+We initially planned to release more frequent major versions, but this one has taken longer than expected. **Upgrading to [MDX v2](https://mdxjs.com/blog/v2/) is the main challenge** to the adoption of this new version. We went the extra mile to make this upgrade as easy as possible, notably by adding [compatibility options for MDX v1](/docs/3.0.0-rc.0/api/docusaurus-config#markdown).
```
Comment (dmitryvinn) [inline website/blog/releases/3.0/index.mdx:23]: We initially planned to release more frequent major versions, but Docusaurus v3 has taken longer than expected.

Comment (dmitryvinn) [inline website/blog/releases/3.0/index.mdx:23]: We might have to update MDX version here, depending on whether we upgrade to v3

Comment (Josh-Cena) [reply website/blog/releases/3.0/index.mdx:23]:
```suggestion
We initially planned to release more frequent major versions, but Docusaurus v3 has taken longer than expected. **Upgrading to [MDX v2](https://mdxjs.com/blog/v2/) is the main challenge** to the adoption of this new version. We went the extra mile to make this upgrade as easy as possible, notably by adding [compatibility options for MDX v1](/docs/3.0.0-rc.0/api/docusaurus-config#markdown).
```

**Comment by Josh-Cena** [comment]
@dmitryvinn To make editorial suggestions, please use the suggestion button, especially because otherwise it's hard to see how the Markdown works.

**Pull request merged**